selfservice/strategy/saml/strategy_registration.go

```diff
@@ -0,0 +1,138 @@
+package saml
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/kratos/identity"
+	"github.com/ory/x/decoderx"
+
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/text"
+
+	"github.com/tidwall/sjson"
+
+	"github.com/ory/kratos/x"
+)
+
+// Implement the interface
+var _ registration.Strategy = new(Strategy)
+
+// Call at the creation of Kratos, when Kratos implement all authentication routes
+func (s *Strategy) RegisterRegistrationRoutes(r *x.RouterPublic) {
+	s.setRoutes(r)
+}
+
+func (s *Strategy) createIdentity(w http.ResponseWriter, r *http.Request, a *registration.Flow, claims *Claims, provider Provider) (*identity.Identity, error) {
+	var jsonClaims bytes.Buffer
+	if err := json.NewEncoder(&jsonClaims).Encode(claims); err != nil {
+		return nil, s.handleError(w, r, a, provider.Config().ID, nil, err)
+	}
+
+	i := identity.NewIdentity(s.d.Config().DefaultIdentityTraitsSchemaID(r.Context()))
+	if err := s.setTraits(w, r, claims, provider, jsonClaims, i); err != nil {
+		return nil, s.handleError(w, r, a, provider.Config().ID, i.Traits, err)
+	}
+
+	s.d.Logger().
+		WithRequest(r).
+		WithField("saml_provider", provider.Config().ID).
+		WithSensitiveField("saml_claims", claims).
+		Debug("SAML Connect completed.")
+	return i, nil
+}
+
+func (s *Strategy) setTraits(w http.ResponseWriter, r *http.Request, claims *Claims, provider Provider, jsonClaims bytes.Buffer, i *identity.Identity) error {
+
+	traitsMap := make(map[string]interface{})
+	json.Unmarshal(jsonClaims.Bytes(), &traitsMap)
+	delete(traitsMap, "iss")
+	delete(traitsMap, "email_verified")
+	delete(traitsMap, "sub")
+	traits, err := json.Marshal(traitsMap)
+	if err != nil {
+		return err
+	}
+	i.Traits = identity.Traits(traits)
```

Comment:
Not sure what this is doing - it is updating the traits of the user with the data from the SAML provider. Is that intentional?